src/Microsoft.DotNet.ProjectModel/WorkspaceContext.cs

```diff
@@ -205,8 +206,23 @@ private FileModelEntry<LockFile> AddLockFileEntry(string projectDirectory, FileM
                 else
                 {
                     currentEntry.FilePath = Path.Combine(projectDirectory, LockFile.FileName);
-                    currentEntry.Model = LockFileReader.Read(currentEntry.FilePath);
-                    currentEntry.UpdateLastWriteTimeUtc();
+
+                    using (var fs = ResilientFileStreamOpener.OpenFile(currentEntry.FilePath, retry: 2))
```

Comment:
The rety window is not for the entire restore, is for the during of actually writing the file.